File: CW/Lista_1/Z3.scala
def replicate[A](x: A, n: Int): List[A] = {
    if(n == 0) {
        List()
    } else if (n > 0) {
        x :: replicate(x, n - 1)
    } else {
        null;
    }
}

@main def main() = {
    val x = "la"
    println(replicate(x, 5))
}

File: CW/Lista_1/Z3.ml
let rec replicate (x: 'a) (n: int): 'a list =
  if n = 0 then
    []
  else if n > 0 then
    x :: replicate x (n - 1)
  else
    failwith "n has to be positive!"

let x = "la";;
assert (replicate x 5 = ["la"; "la"; "la"; "la"; "la"]);;